HiP-DataStore/Core/ReadModel/CacheDatabaseManager.cs

```diff
@@ -83,6 +83,40 @@ private void OnEventAppeared(EventStoreCatchUpSubscription subscription, Resolve
                     _db.GetCollection<Route>(Route.CollectionName).InsertOne(newRoute);
                     break;
 
+                case MediaCreated e:
+                    var newMedia = new MediaElement
+                    {
+                        Id = e.Id,
+                        Title = e.Properties.Title,
+                        Description = e.Properties.Description,
+                        Type = e.Properties.Type,
+                        Status = e.Properties.Status,
+                        IsUsed = false,
+                        File = "",
+                        Timestamp = DateTimeOffset.Now,
+
+                    };
+                    _db.GetCollection<MediaElement>(MediaElement.CollectionName).InsertOne(newMedia);
+                    break;
+
+                case MediaDeleted e:
+                    _db.GetCollection<MediaElement>(MediaElement.CollectionName).DeleteOne(x => x.Id == e.Id);
+                    break;
+                case MediaUpdate e:
+               
+                      var filter = Builders<MediaElement>.Filter.Eq(x => x.Id,e.Id);
+                      var timestamp = new { Timestamp= e.Timestamp }.ToBsonDocument();
+                      var bsonDoc = new BsonDocument("$set", e.Properties.ToBsonDocument().Add(timestamp));
+ 
+                      _db.GetCollection<MediaElement>(MediaElement.CollectionName).UpdateOne(filter,bsonDoc);
+                    break;
+                case MediaFileUpdated e:
+                       var fileDocBson = e.ToBsonDocument();
+                           fileDocBson.Remove("Id");
```

Comment:
fileDocBson.Remove("Id") does not seem to do anything. Did you mean Remove("_id")?